_docs/migration_guides/upgrading_to_terragrunt_0.12.x.md

```diff
@@ -0,0 +1,174 @@
+# Upgrading to Terragrunt 0.12.x
```

Comment:
@brikis98 I created #220, which explains my ideas on moving content from the README into a Terragrunt website. That would eliminate most need for a ToC in the README 👍 

I'll keep an eye on that issue for comments/discussion.